### ISSUE TITLE
Create a standalone 'lizmap config' creator api

### ISSUE THREAD

**Issue by dmarteau** [3liz/lizmap-plugin#101]
For using a Qgis project with Lizmap one has to create a 'Lizmap config file' associated to the project. It could be very handy to have a pyQgis python api that could be used to push new lizmap  project from batch processing  without having to rely on the user interface.

**Comment by mdouchin** [comment]
At present, the creation of the JSON config file has a strong gui dependency. We need to refactor a little bit the plugin to decouple the export of JSON.
Working on it

**Comment by mdouchin** [comment]
@dmarteau please help by giving a smallest python example of loading a QgsProject in QGIS 3 python standalone script. This sample code triggers a segfault in my docker QGIS3 environment ( I am pretty sure I need to add more path definition, perhaps create a QGIS application, etc.)

```
project_path = '/any/path/to/project.qgs'
from qgis.core import QgsProject
p = QgsProject.instance()
p.setFileName(project_path)
print(p.fileName())
p.read(project_path)
```

**Comment by mdouchin** [comment]
Found the issue, I needed to run it as the user defined in my docker container. I will provide a small working python script witch reads a QGIS 3 project

**Comment by dmarteau** [comment]
If you plan to use  pyQgis, you definitely need to to start a QgsApplication, otherwise it could work... or not. 

You have some example code for a full initialization of a qgis 3 application here: https://github.com/3liz/py-qgis-server/blob/master/pyqgisserver/utils/qgis.py

**Comment by mdouchin** [comment]
Thanks @dmarteau It helped. Here is the minimum needed to read the project and get the layers from a python standalone script run inside the `camp2camp/qgis-server` container

```python3
# -*- coding: utf-8 -*-
import sys,os
qgisPrefixPath = "/usr/local/"
sys.path.append(os.path.join(qgisPrefixPath, "share/qgis/python/"))
sys.path.append(os.path.join(qgisPrefixPath, "share/qgis/python/plugins/"))
#os.environ["QGIS_DEBUG"] = '-1'
os.environ['QGIS_PREFIX_PATH'] = qgisPrefixPath

from qgis.core import QgsApplication, QgsProject

# Set the QGIS installation path
QgsApplication.setPrefixPath(qgisPrefixPath, True)
# Init QgsApplication
app = QgsApplication([], False)
# Init QGIS
app.initQgis()

# Read project
project_path = '/home/mdouchin/test_a_sup.qgs'
p = QgsProject()
p.setFileName(project_path)
rp = p.read(project_path)
print(p.mapLayers())

QgsApplication.exitQgis()
app.exit()

```

**Comment by mdouchin** [comment]
@dmarteau new branch `api` with a first basic implementation. See the `README.md`:
https://github.com/3liz/lizmap-plugin/tree/api#api

Implemented: 
* global lizmap options
* options for layers

Missing:
* options for groups
* all the advanced tools : attribute tables, editing, locate, filters, etc.

**Comment by mdouchin** [comment]
@dmarteau Juste improved documentation with 8eca480
https://github.com/3liz/lizmap-plugin/tree/api#api

**Comment by rldhont** [comment]
@dmarteau I have merged #108 

Can this issue be closed ?

**Comment by dmarteau** [comment]
@rldhont yes, the api is functional and in the master branch now.
